tests/all/test_attributes.rs

```diff
@@ -78,56 +80,78 @@ fn test_attributes_on_function_values() {
     let context = Context::create();
     let builder = context.create_builder();
     let module = context.create_module("my_mod");
-    let void_type = context.void_type();
-    let i32_type = context.i32_type();
-    let fn_type = void_type.fn_type(&[i32_type.into()], false);
+    let i32_ptr_type = context.i32_type().ptr_type(AddressSpace::default());
+    let fn_type = i32_ptr_type.fn_type(&[i32_ptr_type.into()], false);
     let fn_value = module.add_function("my_fn", fn_type, None);
     let entry_bb = context.append_basic_block(fn_value, "entry");
     let string_attribute = context.create_string_attribute("my_key", "my_val");
-    let alignstack_attribute = Attribute::get_named_enum_kind_id("alignstack");
-    let enum_attribute = context.create_enum_attribute(alignstack_attribute, 1);
```

Comment:
`alignstack` is not a valid attibute on a function return type. At least for llvm 15, any value beside 0 (1 on line 88 here) leads to issues. The attribute is still applied when that value is 0. I think the value is only useful for some attributes (e.g. `dereferencable(n)`, where `n` is the enum value).

to apply the `noalias` attribute instead, the function return type must  be a pointer type.